.github/workflows/ios-build.yml

```diff
@@ -0,0 +1,46 @@
+name: Build ExecuTorch demo apps
```

Comment:
I think let's not create a new workflow if it's not needed.  We can use https://github.com/pytorch/executorch/blob/main/.github/workflows/app-build.yml with `build-ios` as a different job there